conjure-java-core/src/integrationInput/java/com/palantir/product/ConjureErrors.java

```diff
@@ -55,6 +64,14 @@ public static ServiceException invalidTypeDefinition(Throwable cause, String typ
                 INVALID_TYPE_DEFINITION, cause, SafeArg.of("typeName", typeName), UnsafeArg.of("typeDef", typeDef));
     }
 
+    public static ShouldBeCheckedException shouldBeChecked() {
+        return new ShouldBeCheckedException(SHOULD_BE_CHECKED);
+    }
+
+    public static ShouldBeCheckedException shouldBeChecked(Throwable cause) {
```

Comment:
Not directly related to this PR, but we should update the codegen to annotate these methods with `@CheckReturnValue` so it's obvious calling `shouldBeChecked()` without a `throw` keyword doesn't actually throw.

Reply:
Shall I just tag this onto this PR, or open a separate one?